pantalaimon/daemon.py

```diff
@@ -824,6 +829,60 @@ async def messages(self, request):
             body=await response.read(),
         )
 
+    def _get_upload_and_media_info(self, content_key, content):
+        content_uri = content[content_key]
+
+        try:
+            upload_info = self.upload_info[content_uri]
+        except KeyError:
+            upload_info = self.store.load_upload(self.name, content_uri)
+            if not upload_info:
+                return None, None
+
+        self.upload_info[content_uri] = upload_info
+
+        content_uri = content[content_key]
+        mxc = urlparse(content_uri)
+        mxc_server = mxc.netloc.strip("/")
+        mxc_path = mxc.path.strip("/")
+
+        media_info = self.store.load_media(self.name, mxc_server, mxc_path)
+        if not media_info:
+            return None, None
+
+        self.media_info[(mxc_server, mxc_path)] = media_info
+
+        return upload_info, media_info
+
+    async def _map_decrypted_uri(self, content_key, content, request, client):
+        upload_info, media_info = self._get_upload_and_media_info(content_key, content)
+        if not upload_info or not media_info:
+            raise ValueError
```

Comment:
Let's  use a custom error class for this, just in case something else raises a `ValueError`

Reply:
I added `NotDecryptedAvailableError`. please let me know if how I did is the idiomatic way in python